internal/resources/metal/device/resource.go

```diff
@@ -40,15 +40,15 @@ var (
 	deviceCommonIncludes = []string{"project", "metro", "facility", "hardware_reservation"}
 )
 
-func resourceMetalDevice() *schema.Resource {
+func Resource() *schema.Resource {
 	return &schema.Resource{
 		Timeouts: &schema.ResourceTimeout{
 			Create: schema.DefaultTimeout(30 * time.Minute),
 			Update: schema.DefaultTimeout(30 * time.Minute),
 			Delete: schema.DefaultTimeout(30 * time.Minute),
 		},
 		CreateContext:      resourceMetalDeviceCreate,
-		ReadWithoutTimeout: resourceMetalDeviceRead,
+		ReadWithoutTimeout: Read,
```

Comment:
Why not ReadContext?

Reply:
`ReadWithoutTimeout` and `ReadContext` are both context-aware lifecycle methods: the first argument to the function specified for either of those hooks must be a `context.Context`.  The difference between the two hooks is that `ReadContext` enforces a 20-minute default timeout that can be overridden in a `timeout` block and `ReadWithoutTimeout` doesn't enforce any timeout.  When this is migrated to framework (probably in this PR at some point), we might be forced to revisit timeouts anyway since SDKv2 and framework model/implement them differently.